Engine/Client/Src/Application/Application.cpp

```diff
@@ -61,30 +61,21 @@ void Application::_initDefaultGraphicSystems()
     _defaultSystems.push_back(SpriteSheetAnimationSystem().getFunction());
 }
 
+void Application::_handleAssignTokenPacket(ABINetwork::UDPPacket packet)
+{
+    _idxEntity = ABINetwork::getAssignTokenInfoFromPacket(packet);
+}
+
 void Application::_keyboardHandler(std::size_t key)
 {
     try {
         if (key == KEY_NULL || _client->getToken() == 0)
             return;
 
-        // TODO: Ask Axel to fix that.
-
-        // int idxPlayerPacket = -1;
-        // std::cout << _client->getToken() << std::endl;
-        // ECS::SparseArray<IComponent> PlayerComponentArray = _registry->get_components<IComponent>("PlayerComponent");
-        // for (std::size_t index = 0; index < PlayerComponentArray.size(); index++) {
-        //     std::shared_ptr<PlayerComponent> player = std::dynamic_pointer_cast<PlayerComponent>(PlayerComponentArray[index]);
-        //     if (player)
-        //         std::cout << player->token << std::endl;
-        //     if (player && player->token == _client->getToken()) {
-        //         idxPlayerPacket = index;
-        //         break;
-        //     }
-        // }
-        // if (!_sceneManager->processInput(KEY_MAP(key), idxPlayerPacket))
-        //     return;
-
-        ABINetwork::sendPacketKey(_client, key);
+        if (!_sceneManager->processInput(KEY_MAP(key), _idxEntity))
+            return;
+
+        //ABINetwork::sendPacketKey(_client, key);

```

Comment:
Can you uncomment that please ? :smile: 
We want to send the key to the server.